modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java

```diff
@@ -2821,11 +2821,12 @@ public void remove(GridCacheContext cctx, CacheDataRow row)
      * @param filters Key and value filters.
      * @param <K> Key type.
      * @param <V> Value type.
+     * @param limit Limits response records count. If 0 or less, the limit considered to be Integer.MAX_VALUE, that is virtually no limit.
```

Comment:
Clarification is needed why it is no more than `Integer.MAX_VALUE`? Cannot we have more with `GridCloseableIterator`?

Reply:
Ah, it seems that we use Lucene in that way. How does Lucene interpret `Integer.MAX_VALUE`? Can it return more than `Integer.MAX_VALUE`?